core/c4diagram.go

```diff
@@ -0,0 +1,355 @@
+// package defines diagram as code using the graph input
+package core
+
+import (
+	"bytes"
+	"errors"
+	"io"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+	"unsafe"
+
+	"github.com/kislerdm/diagramastext/core/compression"
+)
+
+type optionsPlantUMLClient struct {
```

Comment:
**question** why use two structs if the HTTP Client is not an optional thing, actually?

Reply:
I'm not sure I've got your question. Could you please clarify. 

For the context, the motivation is extendability and ability to use variadic functions to overwrite defaults. The code for inspiration: [AWS Go SDK](https://github.com/aws/aws-sdk-go-v2/blob/main/config/config.go).

Reply:
Thanks for pointing in the direction of simplicity! I'm going to drop options as unnecessarily complexity. The direction I think to follow:

```go
type clientPlantUML struct {
	baseURL string

        HttpClient HttpClient
}
```